File: backend/app/schemas/project.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ProjectBase(BaseModel):
    title: str
    description: Optional[str] = None
    settings: Optional[str] = None # JSON string for now

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    settings: Optional[str] = None

class ProjectImport(BaseModel):
    file_path: str

class Project(ProjectBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    user_id: Optional[str] = None

    class Config:
        from_attributes = True

File: backend/src/schemas/project.ts
import { z } from 'zod';

export const ProjectSchema = z.object({
  id: z.number().int(),
  title: z.string(),
  description: z.string().nullable().optional(),
  created_at: z.string().or(z.date()),
  updated_at: z.string().or(z.date()).nullable().optional(),
  user_id: z.string().nullable().optional(),
  settings: z.string().nullable().optional() // JSON string
});

export const ProjectCreateSchema = z.object({
  title: z.string(),
  description: z.string().nullable().optional(),
  settings: z.string().nullable().optional()
});

export const ProjectUpdateSchema = z.object({
  title: z.string().optional(),
  description: z.string().nullable().optional(),
  settings: z.string().nullable().optional()
});

export type Project = z.infer<typeof ProjectSchema>;
export type ProjectCreate = z.infer<typeof ProjectCreateSchema>;
export type ProjectUpdate = z.infer<typeof ProjectUpdateSchema>;
